python/fastquant/strategies/base.py

```diff
@@ -318,76 +341,30 @@ def next(self):
-                        self.log("Price: {}".format(self.dataclose[0]))
-                        self.log("Buy prop size: {}".format(buy_prop_size))
-                        self.log("Afforded size: {}".format(afforded_size))
-                        self.log("Final size: {}".format(final_size))
+
                     # Explicitly setting exectype=bt.Order.Close will make the next day's closing the reference price
                     self.order = self.buy(size=final_size)
 
-                    # Implement stop loss at the purchase level (only this specific trade is closed)
                     if self.stop_loss:
                         stop_price = self.data.close[0] * (
                             1.0 - self.stop_loss
                         )
-                        if self.transaction_logging:
-                            self.log("Stop price: {}".format(stop_price))
-                        self.sell(
-                            exectype=bt.Order.Stop,
-                            price=stop_price,
-                            size=final_size,
-                        )
 
-                    if self.stop_trail:
-                        if self.transaction_logging:
-                            self.log("Stop trail: {}".format(self.stop_trail))
-                        self.sell(
-                            exectype=bt.Order.StopTrail,
-                            trailpercent=self.stop_trail,
-                            size=final_size,
-                        )
-
-                # Buy based on the opening price of the next closing day (only works "open" data exists in the dataset)
-                else:
-                    # Margin is required for buy commission
-                    afforded_size = int(
-                        self.cash
-                        / (self.dataopen[1] * (1 + self.commission + 0.001))
-                    )
-                    final_size = min(buy_prop_size, afforded_size)
-                    if self.transaction_logging:
-                        self.log("Buy prop size: {}".format(buy_prop_size))
-                        self.log("Afforded size: {}".format(afforded_size))
-                        self.log("Final size: {}".format(final_size))
-                    self.order = self.buy(size=final_size)
-
-                    # Implement stop loss at the purchase level (only this specific trade is closed)
-                    if self.stop_loss:
-                        stop_price = self.data.close[0] * (
-                            1.0 - self.stop_loss
-                        )
                         if self.transaction_logging:
                             self.log("Stop price: {}".format(stop_price))
-                        self.sell(
-                            exectype=bt.Order.Stop,
-                            price=stop_price,
-                            size=final_size,
-                        )
 
-                    if self.stop_trail:
```

Comment:
I just realized, can we just add back the `self.stop_trail` code block even if it's still problematic? Just so we have some version of it implemented (could be some people using it already).

Reply:
Sure thing i'll add it back again